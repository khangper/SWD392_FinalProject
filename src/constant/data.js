export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
