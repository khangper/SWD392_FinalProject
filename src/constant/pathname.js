export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
