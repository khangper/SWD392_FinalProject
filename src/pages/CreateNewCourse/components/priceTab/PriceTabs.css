.price-course-pricing-tabs {
    width: 100%;
  }
  
  .price-tab-buttons {
    display: flex;
    padding: 30px 0 5px;
  }
  
  .price-tab-button {
    flex: 1;
    border-radius: 0 !important;
    border-top: 1px solid #efefef;
    border-bottom: 1px solid #efefef;
    border-right: 1px solid #efefef;
    border-left: 0;
    font-size: 14px;
    text-align: center;
    font-weight: 500;
    color: #333;
    background: #f7f7f7;
    padding: 12px !important;
    cursor: pointer;
    border: none;
    transition: background 0.3s;
  }
  
  .price-tab-button.price-active {
    background: red;
    color: white;
  }
  
  .price-tab-content {
    padding-top: 35px;
    margin-bottom: 0;
  }
  
  .price-tab-panel {
    display: flex;
    justify-content: center;
  }

.lbl-quiz{
    margin-left: 50px;
    margin-top: 3px;
    font-size: 14px;
    font-weight: bold;
    color: #231f20;
    cursor: pointer;
    top:-2px;
    position: relative;
}
.mt-3{
    margin-top: 1rem !important;
}
.price-tab-block p{
    margin-top: 20px;
    text-align: center;
    padding: 0 150px;
}
.headline-group-2{
    width: 30%;
}