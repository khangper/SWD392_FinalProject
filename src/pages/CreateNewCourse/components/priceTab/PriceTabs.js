import React, { useState } from 'react';
import './PriceTabs.css';

const PriceTabs = ({price, setPrice}) => {
  const [activeTab, setActiveTab] = useState('free');
  const handlePriceChange = (e) => {
    const value = e.target.value.replace(/[^0-9]/g, ''); 
    setPrice(value);
  };
  return (
    <div className="price-course-pricing-tabs">
      <div className="price-tab-buttons">
        <button
          className={`price-tab-button ${activeTab === 'free' ? 'price-active' : ''}`}
          onClick={() => setActiveTab('free')}
        >
          <i className="fas fa-tag"></i> Free
        </button>
        <button
          className={`price-tab-button ${activeTab === 'paid' ? 'price-active' : ''}`}
          onClick={() => setActiveTab('paid')}
        >
          <i className="fas fa-shopping-cart"></i> Paid
        </button>
      </div>
      <div className="price-tab-content">
        {activeTab === 'free' && (
            <div className='price-tab-block'>
                <div className="price-tab-panel">
          <label className="lecture-switch">
                        <input
                          type="checkbox"
                          className="preview-op"
                          value=""
                        />
                        <span></span>
                      </label>
                      <label htmlFor='require-login' className='lbl-quiz'>Require Log in</label>
          </div>
                <div className="price-tab-panel mt-3">
          <label className="lecture-switch">
                        <input
                          type="checkbox"
                          className="preview-op"
                          value=""
                        />
                        <span></span>
                      </label>
                      <label htmlFor='require-login' className='lbl-quiz'>Require Enroll</label>
          </div>
          <p>If the course is free, if student require to enroll your course, if not required enroll, if students required sign in to your website to take this course.</p>
            </div>
          
          
          
        )}
        {activeTab === 'paid' && (
          <div className="price-tab-panel-2">
            <div className="content-tab-title">
                  <h4>Regular Price*</h4>
                </div>
                <div className="headline-group-2">
                    <div className="form-group">
                      <input
                        type="text"
                        id="price"
                        name="price"
                        placeholder="$0"
                        value={price}
                        onChange={handlePriceChange}
                      />
                      <div className="form-counter">USD</div>
                    </div>
                  </div>
                  <div className="content-tab-title">
                  <h4>Discount Price*</h4>
                </div>
                <div className="headline-group-2">
                    <div className="form-group">
                      <input
                        type="text"
                        id="headline"
                        name="headline"
                        placeholder="$0"
                      />
                      <div className="form-counter">USD</div>
                    </div>
                  </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default PriceTabs;
