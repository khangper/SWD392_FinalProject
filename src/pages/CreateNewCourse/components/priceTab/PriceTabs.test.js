import React from "react";
import { render, screen } from "@testing-library/react";
import PriceTabs from "../priceTab/PriceTabs";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../../../constant/test-util";

describe("PriceTabs", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the PriceTabs component", async () => {
    const { container } = renderWithProviders(<PriceTabs />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
