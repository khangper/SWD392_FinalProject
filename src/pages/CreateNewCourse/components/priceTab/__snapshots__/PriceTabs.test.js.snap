// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceTabs renders the PriceTabs component 1`] = `
<div>
  <div
    class="price-course-pricing-tabs"
  >
    <div
      class="price-tab-buttons"
    >
      <button
        class="price-tab-button price-active"
      >
        <i
          class="fas fa-tag"
        />
         Free
      </button>
      <button
        class="price-tab-button "
      >
        <i
          class="fas fa-shopping-cart"
        />
         Paid
      </button>
    </div>
    <div
      class="price-tab-content"
    >
      <div
        class="price-tab-block"
      >
        <div
          class="price-tab-panel"
        >
          <label
            class="lecture-switch"
          >
            <input
              class="preview-op"
              type="checkbox"
              value=""
            />
            <span />
          </label>
          <label
            class="lbl-quiz"
            for="require-login"
          >
            Require Log in
          </label>
        </div>
        <div
          class="price-tab-panel mt-3"
        >
          <label
            class="lecture-switch"
          >
            <input
              class="preview-op"
              type="checkbox"
              value=""
            />
            <span />
          </label>
          <label
            class="lbl-quiz"
            for="require-login"
          >
            Require Enroll
          </label>
        </div>
        <p>
          If the course is free, if student require to enroll your course, if not required enroll, if students required sign in to your website to take this course.
        </p>
      </div>
    </div>
  </div>
</div>
`;
